scripts/buildorder.py

```diff
@@ -262,104 +278,133 @@ def read_packages_from_directories(directories, fast_build_mode, full_buildmode)
-                    yield from find_cycles(deps, dep, path + [pkg])
+def remove_only_installing_deps(pkgs_map, deps):
+    """Complete replacement of packages that have the `TERMUX_{SUB}PKG_ONLY_INSTALLING` (`only_installing` value)
+    variable set to `true` with their dependencies from the `pkg_deps` value."""
 
-        cycles = set()
-        for pkg in remaining_deps:
-            for path_with_cycle in find_cycles(remaining_deps, pkg, []):
-                # Cut the path down to just the cycle.
-                cycle_start = path_with_cycle.index(path_with_cycle[-1])
-                cycles.add(tuple(path_with_cycle[cycle_start:]))
-        for cycle in sorted(cycles):
-            print(f"cycle: {' -> '.join(cycle)}", file=sys.stderr)
+    # list of packages that have been removed from the dependency list,
+    # it is necessary that these packages cannot appear in the dependency list again
+    pkgs_only_installing = set()
 
-        sys.exit(1)
+    while True:
+        boots_only_installing = [pkgs_map[dep].only_installing for dep in deps]
+        if True in boots_only_installing:
+            dep = list(deps)[boots_only_installing.index(True)]
+            pkgs_only_installing |= {dep}
+            deps |= pkgs_map[dep].pkg_deps
+            deps -= pkgs_only_installing
+        else:
+            break
 
-    return build_order
+    return deps
+
+def generate_full_buildorder(pkgs_map, build_mode=True, without_cyclic_dependencies=False):
+    "Generate a build order for building all packages."
+
+    # list that will store the names of packages (with the names of their subpackages) sorted by dependencies
+    pkgs_sort = []
+
+    # dictionary that will store packages and their unfound dependencies in order to find cyclic dependencies
+    requireds = {}
+
+    # copy of the pkgs_map list without subpackages which will contain only unsorted packages
+    pkgs_map_copy = {pkg.name:pkg for pkg in pkgs_map.values() if not build_mode or isinstance(pkg, TermuxPackage)}
+
+    # Start sorting packages by dependencies.
+    while len(pkgs_sort) < len(pkgs_map):
+        # This loop is necessary to repeat the check of package dependencies
+        # with each new content of the `pkgs_sort` list. An infinite loop will
+        # not occur since the checking algorithm will update the `pkgs_sort`
+        # list and there are additional protections in `buildorder.py` that
+        # prevent package dependencies from being configured incorrectly.
+        old_len_sort = len(pkgs_sort)
+        for pkg in pkgs_map_copy.copy().values():
+            subpkgs = [subpkg.name for subpkg in pkg.subpkgs] if build_mode else []
+            # Getting the complete list of package dependencies
+            deps = pkg.deps.copy()
+            if build_mode:
+                for subpkg in subpkgs:
+                    deps |= pkgs_map[subpkg].deps
+                deps = remove_only_installing_deps(pkgs_map, deps-{pkg.name}-set(subpkgs))
```

Comment:
If these are unary minus operators, as the syntax highlighting would imply, then please separate them with spaces.
That way it's clear we're subtracting `{pkg.name}` and `set(subpkgs)` from `deps`
```suggestion
                deps = remove_only_installing_deps(pkgs_map, deps - {pkg.name} - set(subpkgs))
```
If they're not a subtraction, then what the hell is going on here?